src/transformers/trainer.py

```diff
@@ -2384,7 +2385,10 @@ def compute_loss(self, model, inputs, return_outputs=False):
             self._past = outputs[self.args.past_index]
 
         if labels is not None:
-            loss = self.label_smoother(outputs, labels)
+            if model._get_name() in MODEL_FOR_CAUSAL_LM_MAPPING_NAMES.values():
```

Comment:
The `model` might need to be unwrapped here to deal with distributed settings.

Reply:
Hi, thanks for the words. As you said, I've applied `make style`.
About the unwrapping `model` part, could you please elaborate? 
As for now, it seems there is no problem because the code passed all the tests.

Reply:
Yes, the distributed tests are not run on each commit, so I would expect them to pass. You need to use the function `unwrap_model` (already imported in this module) or you might get `DistributedDataParallel` instead of your model name:
```suggestion
            if unwrap_model(model)._get_name() in MODEL_FOR_CAUSAL_LM_MAPPING_NAMES.values():
```

Reply:
Ah! Thanks for the explanation. I understood. 
I thought maybe just committing your suggestion is enough, so I did it.